### PR TITLE
[TIMOB-14382] Android: Allow selecting multiple photos with Ti.Media.openPhotoGallery

### PR DESCRIPTION
**Pull request by manumaticx**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-14382

Test code in JIRA

### CODE REVIEW

File: apidoc/Titanium/Media/Media.yml [hunk 1]
```diff
@@ -1669,6 +1669,10 @@ properties:
     type: Number
     constants: Titanium.UI.iPad.POPOVER_ARROW_DIRECTION*
     platforms: [ipad]
+  - name: allowMultiple
+    summary: Specifies if the user should be able to select multiple photos.
+    type: Boolean
+    platforms: [android]
```
Comment (hansemannn) [inline apidoc/Titanium/Media/Media.yml]: Missing `since` version. Since we are pretty settled for the 5.4.0 feature freeze, it likely goes into 6.0.0. @ashcoding decides!

**Comment by ashcoding** [comment]
If possible, I'll want to get it in 5.4.0.

I'll need to review and test this.

**Comment by hansemannn** [comment]
Please do so, that's even better! 👍

File: android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java [hunk 2]
```diff
@@ -910,12 +912,22 @@ public void openPhotoGallery(KrollDict options)
 		TiActivitySupport activitySupport = (TiActivitySupport) activity;
 
 		TiIntentWrapper galleryIntent = new TiIntentWrapper(new Intent());
-		galleryIntent.getIntent().setAction(Intent.ACTION_PICK);
+		galleryIntent.getIntent().setAction(Intent.ACTION_GET_CONTENT);
```
Comment (manumaticx) [inline android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java]: btw. this is not a necessary change but recommended by Google since `ACTION_PICK` can be considered as deprecated (http://stackoverflow.com/questions/6486716/using-intent-action-pick-for-specific-path/6486827#6486827)

Comment (ashcoding) [inline android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java]: Thank you for the explanation.

**Comment by ashcoding** [comment]
@manumaticx I'm trying to review this to put into 6.0.0.

Can't find the test code that is said to be in Jira. Could you provide test code for this?

File: apidoc/Titanium/Media/Media.yml [hunk 3]
```diff
@@ -1669,6 +1669,10 @@ properties:
     type: Number
     constants: Titanium.UI.iPad.POPOVER_ARROW_DIRECTION*
     platforms: [ipad]
+  - name: allowMultiple
+    summary: Specifies if the user should be able to select multiple photos.
```
Comment (ashcoding) [inline apidoc/Titanium/Media/Media.yml]: Can you mention this is available only on API 18 and above?

**Comment by manumaticx** [comment]
@ashcoding: Testcode as posted in [TIMOB-23168](https://jira.appcelerator.org/browse/TIMOB-23168):

```js
var win = Ti.UI.createWindow({
  backgroundColor: '#fff',
  layout: 'vertical'
});
 
var button = Ti.UI.createButton({
  title: 'open gallery'
});
 
button.addEventListener('click', function(){
  Ti.Media.openPhotoGallery({
    allowMultiple: true,
    success: function(e){
      e.images.forEach(function(image){
        var imageView = Ti.UI.createImageView({
          height: 300,
          image: image.media
        });
        win.add(imageView);
      });
    }
  });
});
 
win.add(button);
 
win.open();
```

File: apidoc/Titanium/Media/Media.yml (hunk 3 above)
Comment (manumaticx) [inline apidoc/Titanium/Media/Media.yml]: How do I declare the API version in docs?

Comment (hansemannn) [inline apidoc/Titanium/Media/Media.yml]: Just write it into the `description` as an additional information for the user.

**Comment by ashcoding** [comment]
Thanks for the testcode and PR.

Tried to functionally test this using a Galaxy S5 (SM-G900F) running Android 4.4.2 and a Nexus 6 running Android 6.0. Not able make it work.

Able to open into Photos app and select multiple images. Once back into the app, nothing is seen.

**Comment by ashcoding** [comment]
@manumaticx Doesn't seem to be working correctly for me. The images don't appear with the test code.

**Comment by ashcoding** [comment]
There's a few edge cases and some issue with the logic that's not being handled. I'm going to look into it.

**Comment by ashcoding** [comment]
It doesn't work well. @build

**Comment by ashcoding** [comment]
@manumaticx there were changes that I had to do to make it work. I hope you don't mind me taking your code and adding the modifications needed.

That PR is here. https://github.com/appcelerator/titanium_mobile/pull/8081
Thank you for your initial PR.

Will be closing this PR in favour of https://github.com/appcelerator/titanium_mobile/pull/8081.

**Comment by manumaticx** [comment]
@ashcoding thanks a lot for fixing this! :)
I just didn't had the time to have a look at this again atm
